### PR TITLE
Sentry fix for new update_code strategies

### PR DESCRIPTION
**Pull request by dpfaffenbauer**
- [x] Bug fix #…?

Sentry does not work properly with the new `update_code_strategy`, this fixes it to work with both

### CODE REVIEW

File: contrib/sentry.php [hunk 1]
```diff
@@ -192,14 +188,49 @@ static function (&$value) use ($config) {
...
 }
 
 function getReleaseGitRef(): Closure
 {
     return static function ($config = []): string {
+        if (get('update_code_strategy') === 'archive') {
+            return run('cat {{current_path}}/REVISION');
+        }
+
         cd('{{release_path}}');
```
Comment (DamienHarper) [inline contrib/sentry.php:234]: Same here, you'd better `cd('{{deploy_path}}/.dep/repo');` instead to be able to run custom git command with our having to rely on `update_code_strategy => 'clone'`

File: contrib/sentry.php [hunk 2]
```diff
@@ -192,14 +188,49 @@ static function (&$value) use ($config) {
...
+}
+
+function getCurrentReleaseRevision()
+{
+    switch (get('update_code_strategy')) {
+        case 'archive':
+            return run('cat {{release_path}}/REVISION');
+
+        case 'clone':
+            cd('{{release_path}}');
```
Comment (DamienHarper) [inline contrib/sentry.php:219]: You'd better cd('{{deploy_path}}/.dep/repo'); instead to be able to run custom git command with our having to rely on update_code_strategy => 'clone'

**Review by DamienHarper — CHANGES_REQUESTED** [review]

File: contrib/sentry.php [hunk 3]
```diff
@@ -192,14 +188,49 @@ static function (&$value) use ($config) {
...
+        default:
+            throw new ConfigurationException(parse("Unknown `update_code_strategy` option: {{update_code_strategy}}."));
+    }
+}
+
+function getCurrentReleaseRevision()
+{
+    switch (get('update_code_strategy')) {
+        case 'archive':
+            return run('cat {{release_path}}/REVISION');
```
Comment (DamienHarper) [inline contrib/sentry.php:216]: if a custom `git_version_command` has been provided, you'd better `cd('{{deploy_path}}/.dep/repo');` then run the custom `git_version_command` instead.

File: contrib/sentry.php [hunk 4]
```diff
@@ -192,14 +188,49 @@ static function (&$value) use ($config) {
...
+        default:
+            throw new ConfigurationException(parse("Unknown `update_code_strategy` option: {{update_code_strategy}}."));
+    }
 }
 
 function getReleaseGitRef(): Closure
 {
     return static function ($config = []): string {
+        if (get('update_code_strategy') === 'archive') {
+            return run('cat {{current_path}}/REVISION');
```
Comment (DamienHarper) [inline contrib/sentry.php:231]: Same here, if a custom `git_version_command` has been provided, you'd better `cd('{{deploy_path}}/.dep/repo');` then run the custom `git_version_command` instead.

**Review by DamienHarper — CHANGES_REQUESTED** [review]

**Comment by antonmedv** [comment]
Why not use REVISION file in all cases?

**Comment by DamienHarper** [comment]
@antonmedv because I don't want only a commit hash as a version identifier. I used to use `git describe --tags --abbrev=0 2> /dev/null || git log -n 1 --format="%h"` as a custom `git_version_command` to get either the latest tag in the branch or the latest commit hash

**Comment by antonmedv** [comment]
I see. Can someone also take a look at this pr? Or I!m going to merge it.

**Comment by DamienHarper** [comment]
@dpfaffenbauer Would you mind to integrate requested changes?

**Comment by dpfaffenbauer** [comment]
@DamienHarper sure

**Comment by dpfaffenbauer** [comment]
@antonmedv so should I create a new PR for requested changes?

**Comment by antonmedv** [comment]
Let’s do separate PR.

**Comment by DamienHarper** [comment]
@antonmedv @dpfaffenbauer I made a PR for this: https://github.com/deployphp/deployer/pull/2793

**Pull request merged**